selfdrive/ui/paint.cc

```diff
@@ -328,23 +328,12 @@ static void ui_draw_vision_maxspeed(UIState *s) {
 }
 
 static void ui_draw_vision_speed(UIState *s) {
-  const Rect &viz_rect = s->scene.viz_rect;
-  float v_ego = s->scene.controls_state.getVEgo();
-  float speed = v_ego * 2.2369363 + 0.5;
-  if (s->is_metric){
-    speed = v_ego * 3.6 + 0.5;
-  }
-  const int viz_speed_w = 280;
-  const int viz_speed_x = viz_rect.centerX() - viz_speed_w/2;
+  const float speed = s->scene.controls_state.getVEgo() * (s->is_metric ? 3.6 : 2.2369363);
   char speed_str[32];
-
-  nvgBeginPath(s->vg);
-  nvgRect(s->vg, viz_speed_x, viz_rect.y, viz_speed_w, header_h);
+  snprintf(speed_str, sizeof(speed_str), "%d", (int)std::rint(speed));
```

Comment:
Maybe use `std::nearbyint` since that one doesn't raise an exception? It's not entirely clear to me in what cases rint raised `FE_INEXACT`.

Maybe also use `std::format` instead of snprintf?

Reply:
std::format is in C++ 20,we can use std:to_string in c++11. but it has one more memory allocation than snprintf.

Reply:
Let's use std::to_string. Those few allocated bytes are negligible at 20 fps.